### ISSUE TITLE
[go_router] Deep link on desktop

### ISSUE THREAD

**Issue by Kiura** [flutter/flutter#134264]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Use case

I thought deep-linking was supported on all platforms because GoRouter claims to support all platforms on pub.dev. I'm looking for a solution to handle deep-linking on all platforms, and the only option I found is to use GoRouter along with [app_links](https://pub.dev/packages/app_links). It's a bit inconvenient to use two libraries for the same purpose, especially when their approaches differ significantly.

### Proposal

It would be nice to cover all deep-linking-related tasks in one library.
I understand that it might be required to do some things manually here and there. At the very least deep-linking handling behavior would be predictable and consistent.
If that's not possible it would be nice to have a heads up in the documentation.

**Comment by dam-ease** [comment]
Hi @Kiura. Thanks for proposing this. Looking at the documentation on deep linking, I can't see it supported on all platforms as you stated, see https://pub.dev/documentation/go_router/latest/topics/Deep%20linking-topic.html. Can you point me to more information on this?
Also, using [app_links](https://pub.dev/packages/app_links/changelog) should suit this particular use case well.

**Comment by Kiura** [comment]
Hi @dam-ease. Thx for the quick reply. I agree, the documentation for deep linking does not state support for desktop platforms. I was referring to the support of all platforms by the package, not just on deep linking.
<img width="546" alt="image" src="https://github.com/flutter/flutter/assets/5522096/8d2bf869-dd06-49cb-8a28-2541178487e9">

So I am guessing, there are no plans to support Desktop platforms by the library, right?

app_links does solve this particular issue, although it seems like the library has limited support for Linux.

**Comment by dam-ease** [comment]
Thanks for your response. Since app_links solves this, I'd suggest using this but I'd label for more insights from the team. Thank you!

**Comment by chunhtai** [comment]
The desktop embedding need to be able to send deeplink request to the framework side, after that it should just work.

**Comment by loic-sharma** [comment]
For Windows, deep links is a bit funky:

1. The app needs to add a registry key for the app's custom scheme (like `myapp://`)
2. When the user opens the deep link, Windows launches a new process for the app...
    1. If this is the app's first process, the newly launched process should navigate to the deep link
    2. If there are already other running processes for the app...
        1. The newly launched process should pick an existing process and notify it of the deep link
        2. The newly launched process should exit

**Comment by chunhtai** [comment]
It looks like for app_links package there is some manual set up in windows, is this something we should add to windows embedding?

**Comment by loic-sharma** [comment]
> It looks like for app_links package there is some manual set up in windows, is this something we should add to windows embedding?

Yup, we'll likely want something like that so that the process launched by the deep link can notify the app's existing process of the deep link.

**Comment by Kiura** [comment]
@loic-sharma, @chunhtai
Having to do some manual setup is fine, I had to do this for app_links anyways + extra code in the initialization file.

So, it would be nice to use go_router for all platforms instead of handling special cases separately. Plus, there would be no need for an extra library.
